exporter/exporterhelper/internal/queuebatch/persistent_queue.go

```diff
@@ -241,6 +346,11 @@ func (pq *persistentQueue[T]) unrefClient(ctx context.Context) error {
 func (pq *persistentQueue[T]) Offer(ctx context.Context, req T) error {
 	pq.mu.Lock()
 	defer pq.mu.Unlock()
+
+	for pq.sizerTypeMismatch {
+		pq.logger.Debug("Blocking new requests due ti sizer type mismatch, waiting for drain to complete")
+		pq.drainingComplete.Wait()
```

Comment:
Can you please explain why you need a cond var as well as a boolean? Why not exit here?